.circleci/config.yml

```diff
@@ -23,6 +23,55 @@ executors:
 
 # Custom commands definition
 commands:
+  # Based on https://circleci.com/developer/orbs/orb/kimh/run-with-retry
+  retry_until_success:
+    description: Retry command multiple times until it succeeds
+    parameters:
+      command:
+        type: string
+      command_name:
+        type: string
+      working_directory:
+        type: string
+      retry_count:
+        type: integer
+        default: 3
+      coverage:
+        type: boolean
+        default: false
+      compat:
+        type: boolean
+        default: false
+      disable_synthetic:
+        type: boolean
+        default: false
+      force_native_shadow_mode:
+        type: boolean
+        default: false
```

Comment:
Nitpick: It feels strange that the generic `retry_until_success` command accepts `coverage`, `compat`, `disable_synthetic`, and `force_native_shadow_mode` as parameters since those parameters are only applicable to the Karma tests. 

IMO it would be preferable to keep them inside the Kama command. It appears the CircleCi config syntax doesn't accept passing arbitrary objects. Many passing them as via a generic `environment` parameter would do the trick.

Reply:
I considered using `environment`, but it's tricky because right now we are relying on the presence or absence of environment variables. E.g. I'm not sure if passing `COVERAGE=0` will actually _disable_ coverage – I suspect it would be converted to a string `"0"` in JavaScript, which is truthy, so coverage would be enabled. We would have to audit each of our environment variables to make sure they handle `"0"`, `"false"`, etc.

I agree the name `retry_until_success` is not great. Maybe `retry_karma_until_success`? Or just `retry_karma`?

Reply:
> Maybe `retry_karma_until_success`? Or just `retry_karma`?

`retry_karma` sounds good. 

> We would have to audit each of our environment variables to make sure they handle `"0"`, `"false"`, etc.

You can set those environment variables to a nullish value like an empty string, which would coerce to a `false` boolean.

```
COVERAGE="" COMPAT="" yarn test
```

While looking into this, I learned that bash allows you to define environment variables without any value. The following code would also work.

```
COVERAGE= COMPAT= yarn test
```

Again it's a nitpick.